src/EmbeddedManifold.jl

```diff
@@ -112,10 +122,10 @@ base_manifold(M::EmbeddedManifold, d::Val{N} = Val(-1)) where {N} = M.manifold
     check_manifold_point(M::AbstractEmbeddedManifold, p; kwargs)
 
 check whether a point `p` is a valid point on the [`AbstractEmbeddedManifold`](@ref),
-i.e. that `embed(M, p)` is a valid point on the embedded manifold.
+i.e. that `project(M, p)` is a valid point on the embedded manifold.
 """
 function check_manifold_point(M::AbstractEmbeddedManifold, p; kwargs...)
-    q = embed(M, p)
+    q = project(M, p)
```

Comment:
maybe this should not be necessary at all, since we assume that `M.manifold` itself does not change, but just the type of embed/projct you use?

Reply:
But well, one actually can check that `embed(...)` yields a valid point on `M.embedding` for sure.

Reply:
Let's assume we have R^2 embedded in R^3. `p` is supposed to be a 3-element vector. We want to convert it to the 2-element vector representation expected by the `check_manifold_point` method invoked below. So that looks like what `project` does I think. I only touched this part because otherwise tests were failing and it sort of makes sense.

Reply:
My interpretation until now was (and that'S also where the Euclidean embeddings come from), that a point `p` on the manifold (R^2 in your example) can have different embeddings we would like to transfer it to (one of them is R^3) but it might be in _any_ representation (depending on which manifold you have).

For example one could think of M being `M=Symmetric(2)` – that can be nicely embedded in N=R3 but points p are 2x2 matrices. We only know
- check_manifold_point(E.Manifold,p) is true
- check_manifold_point(E.embedding, embed(E,p)) is true

Embedded manifold `E(M,N)` acts like `M` but tells us where we embed into if we call embed. (or reversely) where we project from) but it does not change `p`s from `M` itself. It “decorates” `M` with an embedding – but only embeds points when we call `embed`.

Reply:
Yes, you are right. Points are by default stored in the `E.manifold` representation and can be converted using `embed` to the `E.embedding` representation. I will fix this.

Reply:
Still great to discuss, since before I wasn't that sure either, but this way it feels correct.